server/src/services/s3-bucket.ts

```diff
@@ -0,0 +1,35 @@
+import AWS from 'aws-sdk';
+import { ItemWithCategory } from "../db/dal/items.dal";
+import { Item } from "../db/schemas/item.schema";
+
+AWS.config.update({
+    accessKeyId: process.env.S3_ACCESS_KEY_ID,
+    secretAccessKey: process.env.S3_SECRET_ACCESS_KEY,
+    region: 'eu-north-1',
+    signatureVersion: 'v4',
+});
+const s3 = new AWS.S3();
+
+export enum Action {
+    upload = "putObject",
+    download = "getObject",
+}
+
+export const getS3SignedUrl = async (imageName : string, action : Action = Action.download) : Promise<string> => {
+    const url = await s3.getSignedUrlPromise(action, {
```

Comment:
Can't you just save the item with the full imageUrl from S3 in DB?
In this way you don't need to call `getAllItemsWithS3Images` everywhere (e.g. `items.dal.ts` and `favorites.dal.ts`) and you won't need to care in the future if the url is hard-coded or from S3.

Reply:
![image](https://user-images.githubusercontent.com/48032069/236647644-741de9c3-5813-4aee-b537-c186ba68a50f.png)

Reply:
https://www.quora.com/How-can-I-make-a-permanent-URL-for-a-file-Ive-uploaded-to-S3